src/librustc/mir/interpret/allocation.rs

```diff
@@ -509,28 +588,66 @@ impl<'tcx, Tag: Copy, Extra> Allocation<Tag, Extra> {
 
 /// Undefined bytes
 impl<'tcx, Tag, Extra> Allocation<Tag, Extra> {
+    #[inline]
+    fn check_defined(&self, ptr: Pointer<Tag>, size: Size) -> InterpResult<'tcx> {
+        match &self.alloc {
+            Some(alloc) => alloc.check_defined(ptr.offset, size),
+            None if size == Size::ZERO && ptr.offset <= self.size => Ok(()),
+            None => err!(ReadUndefBytes(ptr.offset)),
+        }
+    }
+
+    /// Directly query the UndefMask for a range.
+    /// Does not go through any interpreter hooks so only use for debugging.
+    pub fn check_raw_range_defined(&self, start: Size, end: Size) -> Result<(), Size> {
+        match &self.alloc {
+            Some(alloc) => alloc.undef_mask.is_range_defined(start, end),
+            None if start == end && end <= self.size => Ok(()),
+            None => Err(start),
+        }
+    }
+
+    pub fn mark_definedness(
+        &mut self,
+        ptr: Pointer<Tag>,
+        size: Size,
+        new_state: bool,
+    ) {
+        let alloc = self.force_allocation();
+        alloc.mark_definedness(ptr.offset, size, new_state)
+    }
+
+    /// If an all undef allocation was elided allocate now. Does nothing otherwise.
+    pub fn force_allocation(&mut self) -> &mut AllocationBytes {
```

Comment:
We also have a `force_allocation` method in the Miri engine in `librsutc_mir` and it is very different. Can we avoid using the same name for two very different operations?